src/test/kotlin/org/jmailen/gradle/kotlinter/functional/WithGradleTest.kt

```diff
@@ -1,16 +1,31 @@
 package org.jmailen.gradle.kotlinter.functional
 
+import org.apache.commons.io.FileUtils
 import org.gradle.internal.classpath.DefaultClassPath
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.GradleRunner
 import org.gradle.testkit.runner.internal.PluginUnderTestMetadataReading
-import org.junit.Rule
-import org.junit.rules.TemporaryFolder
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import java.io.File
+import java.nio.file.Files
 
 abstract class WithGradleTest {
 
-    @get:Rule
-    val testProjectDir = TemporaryFolder()
+    lateinit var testProjectDir: File
+
+    /**
+     * Not using JUnit's @TempDir, due do https://github.com/gradle/gradle/issues/12535
+     */
+    @BeforeEach
+    internal fun setUpTempdir() {
+        testProjectDir = Files.createTempDirectory(this::class.java.simpleName).toFile()
+    }
+
+    @AfterEach
+    internal fun cleanUpTempdir() {
+        FileUtils.forceDeleteOnExit(testProjectDir)
+    }
```

Comment:
Normally, I'd replace `get:Rule` with a `@TempDir` annotation, but unfortunately, it seems like JUnit5 tries to delete temp files at a different time than JUnit4 which doesn't play well with test-kit behavior. I applied one of the workarounds I found in the linked issue